streams/src/main/java/org/apache/kafka/streams/kstream/KTable.java

```diff
@@ -719,19 +708,8 @@ void to(final Serde<K> keySerde,
      * <td></td>
      * </tr>
      * </table>
-     * Both input streams need to be co-partitioned on the join key.
-     * If this requirement is not met, Kafka Streams will automatically repartition the data, i.e., it will create an
-     * internal repartitioning topic in Kafka and write and re-read the data via this topic before the actual join.
-     * The repartitioning topic will be named "${applicationId}-XXX-repartition", where "applicationId" is
```

Comment:
We don't get any repartitioning topics for KTables, thus this explanation does not apply.

It's a KTable invariant that it is partitioned by key -- thus, if you "feed" it into a join, you know that it's partitioned correctly (this was a c&p error for KStream-KStream joins).